activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -137,6 +137,10 @@ def convert_dot_notation_to_hash(attributes)
       def handler_for(object)
         @handlers.detect { |klass, _| klass === object }.last
       end
+
+      def association_stuff?(value)
+        value.is_a?(Array) || value.is_a?(Base) || value.is_a?(Relation) || value.nil?
```

Comment:
What do you think of using `any?` instead of `||` for this?

```ruby
p [Array, Base, Relation, NilClass].any? { |klass| [].is_a?(klass) } # true
p [Array, Base, Relation, NilClass].any? { |klass| false.is_a?(klass) } # false
```

Reply:
I see.
This is DRY. Thanks